pymeasure/instruments/channel.py

```diff
@@ -109,6 +109,20 @@ def check_errors(self):
         """
         return self.parent.check_errors()
```

Comment:
I think I agree with your reasoning w.r.t. methods calling the parent's counterparts. However, what do we use `Channel.check_errors` for, now? The Channel's get/set variants call the parent's versions directly, so don't need that, also not in the default case.

Reply:
> However, what do we use Channel.check_errors for, now?

Mainly for backwards compatibility, in case someone calls `check_errors` in a method.

Shall we deprecate it? Do we have to deprecate it (Channels are not yet in a release)?

There could be a hypothetical instrument, where you want to ask channel errors with `check_errors`, but that instrument may implement that method anyways.

Reply:
I checked the codebase. The following instruments use `check_errors` in a Channel, intended to call the parent's one:

- Keithley2600 (not yet `Channel` class)
- AgilentB1500 (not yet `Channel` class)
- Agilent4156 (not yet `Channel` class)
- KeithleyBuffer (not yet `Channel` class)

In all these occurrences, a method calls `check_errors` in ordert o **ask** for errors. They could use the `check_get/set_errors` methods instead, but it does not seem right to me, as they are intended for getters/setters.

Therefore, I propose to keep all three methods in the Channel.

Reply:
> Therefore, I propose to keep all three methods in the Channel.

Your reasoning sounds good, I agree.

> Do we have to deprecate it (Channels are not yet in a release)?

Generally, if something hasn't landed in a published release, yet, there is no need to follow a deprecation process, just change it as needed. That is partly the idea with doing a release _before_ merging a big feature -- there is more time to iron out any problems using the master branch.